agentendpoint/agentendpoint.go

```diff
@@ -199,19 +271,27 @@ func (c *Client) loadTaskFromState(ctx context.Context) error {
 
 // WaitForTaskNotification waits for and acts on any task notification indefinitely.
 func (c *Client) WaitForTaskNotification(ctx context.Context) error {
+	logger.Debugf("Checking local state file for saved task.")
 	if err := c.loadTaskFromState(ctx); err != nil {
 		logger.Errorf(err.Error())
 	}
 
+	logger.Debugf("Setting up ReceiveTaskNotification stream watcher.")
 	noti := make(chan struct{}, 1)
 	for {
 		token, err := config.IDToken()
 		if err != nil {
-			return err
+			return fmt.Errorf("error fetching Instance IDToken: %v", err)
 		}
 
 		if err := c.watchStream(ctx, noti, token); err != nil {
-			return err
+			if s, ok := status.FromError(err); ok && s.Code() == codes.PermissionDenied {
+				// Service is not enabled for this project.
+				time.Sleep(config.SvcPollInterval())
+				continue
+			}
+			logger.Errorf("Error watching stream: %v", err)
+			time.Sleep(5 * time.Second)
```

Comment:
Can you add a TODO to add backoff? This could be problematic without it.

Reply:
Done, this can get complicated, we may want to add different waits for many different classes of errors.